pennylane/beta/tapes/tape.py

```diff
@@ -936,6 +1012,12 @@ def execute_device(self, params, device):
 
         # restore original parameters
         self.set_parameters(saved_parameters)
+
+        if self._caching and hashed_params not in self._cache_execute:
+            self._cache_execute[hashed_params] = res
+            if len(self._cache_execute) > self._caching:
+                self._cache_execute.popitem(last=False)
```

Comment:
It would be good to document this behaviour: users could also think that once the `caching` limit is reached, then nothing happens to the cache. However, this seems to be not the case as each time there's a new execution, the very first cached result is dropped and we're adding the latest result.

Overall also might be worth considering which of the two options is more beneficial.